### ISSUE TITLE
Error building with BOX32 option on Neoverse-N1 aarch64/Oracle Cloud (error: ‘FUTEX_LOCK_PI2’ undeclared (first use in this function); did you mean ‘FUTEX_LOCK_PI’?)

### ISSUE THREAD

**Issue by VYRNACH** [ptitSeb/box64#2138]
Tried building using various options combinations with cmake .. -D **BOX32=1** and it all returned this specific error no matter what I try.

`/home/ubuntu/box64/src/emu/x86syscall_32.c:350:26: error: ‘FUTEX_LOCK_PI2’ undeclared (first use in this function); did you mean ‘FUTEX_LOCK_PI’?
  350 |                     case FUTEX_LOCK_PI2:
      |                          ^~~~~~~~~~~~~~
      |                          FUTEX_LOCK_PI
/home/ubuntu/box64/src/emu/x86syscall_32.c:350:26: note: each undeclared identifier is reported only once for each function it appears in
/home/ubuntu/box64/src/emu/x86syscall_32.c: In function ‘my32_syscall’:
/home/ubuntu/box64/src/emu/x86syscall_32.c:533:26: error: ‘FUTEX_LOCK_PI2’ undeclared (first use in this function); did you mean ‘FUTEX_LOCK_PI’?
  533 |                     case FUTEX_LOCK_PI2:
      |                          ^~~~~~~~~~~~~~
      |                          FUTEX_LOCK_PI
make[2]: *** [CMakeFiles/box64.dir/build.make:1297: CMakeFiles/box64.dir/src/emu/x86syscall_32.c.o] Error 1
`

Cpu info (if relevant):
Architecture:                         aarch64
CPU op-mode(s):                       32-bit, 64-bit
Byte Order:                           Little Endian
CPU(s):                               4
On-line CPU(s) list:                  0-3
Thread(s) per core:                   1
Core(s) per socket:                   4
Socket(s):                            1
NUMA node(s):                         1
Vendor ID:                            ARM
Model:                                1
Model name:                           Neoverse-N1
Stepping:                             r3p1
BogoMIPS:                             50.00
NUMA node0 CPU(s):                    0-3
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; CSV2, BHB
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected
Flags:                                fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp

OS details:
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal

Sorry if this is a bad report as I'm kinda new to this.
Possibly offtopic: Forced to builld as the GitHub runner is using GLIBC 2.32 while my Ubuntu is using 2.31
/usr/local/bin/box32: /lib/aarch64-linux-gnu/libm.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/box32)

**Comment by VYRNACH** [comment]
that definitely fixed it! Thanks for the quick response!

Possibly offtopic: but I've gotten another error building with -D ADLINK=1 -D ARM_DYNAREC=ON -D CMAKE_BUILD_TYPE=RelWithDebInfo -D BOX32=1. To make sure it isn't ADLINK and DYNAREC, I also ran -D CMAKE_BUILD_TYPE=RelWithDebInfo -D BOX32=1 and got the same error
`[ 95%] Building C object CMakeFiles/box64.dir/src/wrapped32/wrappedlibresolv.c.o
In file included from /home/ubuntu/box64/src/wrapped32/wrappedlibresolv.c:28:
/home/ubuntu/box64/src/wrapped32/generated/wrappedlibresolvtypes32.h:18:5: error: duplicate member ‘__res_query’
   18 |  GO(res_query, iEpiipi_t)
      |     ^~~~~~~~~
/home/ubuntu/box64/src/wrapped32/wrappercallback32.h:7:29: note: in definition of macro ‘GO’
    7 |     #define GO(A, B)    B   A;
      |                             ^
/home/ubuntu/box64/src/wrapped32/wrappercallback32.h:8:5: note: in expansion of macro ‘SUPER’
    8 |     SUPER()
      |     ^~~~~
make[2]: *** [CMakeFiles/box64.dir/build.make:4822: CMakeFiles/box64.dir/src/wrapped32/wrappedlibresolv.c.o] Error 1
`
Should I make another GitHub Issue for this?

**Comment by ptitSeb** [comment]
Yeah, I would prefer a new ticket as it's a completly different error.

**Comment by VYRNACH** [comment]
Original error resolved with thanks

> that definitely fixed it! Thanks for the quick response!
> 
> Possibly offtopic: but I've gotten another error building with **-D ADLINK=1 -D ARM_DYNAREC=ON -D CMAKE_BUILD_TYPE=RelWithDebInfo -D BOX32=1**. To make sure it isn't ADLINK and DYNAREC, I also ran **-D CMAKE_BUILD_TYPE=RelWithDebInfo -D BOX32=1** and got the same error `[ 95%] Building C object CMakeFiles/box64.dir/src/wrapped32/wrappedlibresolv.c.o In file included from /home/ubuntu/box64/src/wrapped32/wrappedlibresolv.c:28: /home/ubuntu/box64/src/wrapped32/generated/wrappedlibresolvtypes32.h:18:5: error: duplicate member ‘__res_query’ 18 | GO(res_query, iEpiipi_t) | ^~~~~~~~~ /home/ubuntu/box64/src/wrapped32/wrappercallback32.h:7:29: note: in definition of macro ‘GO’ 7 | #define GO(A, B) B A; | ^ /home/ubuntu/box64/src/wrapped32/wrappercallback32.h:8:5: note: in expansion of macro ‘SUPER’ 8 | SUPER() | ^~~~~ make[2]: *** [CMakeFiles/box64.dir/build.make:4822: CMakeFiles/box64.dir/src/wrapped32/wrappedlibresolv.c.o] Error 1 ` Should I make another GitHub Issue for this?

#Issue 2140 created for this separate error
